### DISCUSSION TITLE
Multiple self service user interfaces

### DISCUSSION THREAD

**Discussion by openscript** [ory/kratos#3395] [Q&A] [answered]
Is it possible that a Kratos instance have multiple self service UIs? The use case is the following. We have a native app, but also a browser version of it. How can we attach once the native self service and the browser self service to a single Kratos instance?

**Comment by Benehiko** [comment, marked as answer]
Hi @openscript 

The native flows just call the `api` endpoints while the web app uses the `browser` endpoints. Browser apps usually redirect the user between flows (except for SPAs), while native apps only use ajax JSON requests.

 https://www.ory.sh/docs/identities/native-browser
 
Both will work out of the box on the same instance of Kratos. The only thing that is not supported in Kratos directly, is having multiple flow UIs set for browser applications.
e.g. login on `https://app1.com/login` and `https://app2.com/login2`. 

For that you need to spin up multiple Kratos instances and manage different configuration files. Ory Network, however, supports multiple custom domains on a single project. See this document https://www.ory.sh/docs/kratos/bring-your-own-ui/configure-ory-to-use-your-ui

**Reply by openscript** [reply]
Thank you @Benehiko for taking the time and answer my question.

I've just learned about the `return_to` parameter for login and registration flows and I agree with this we can have both a native app and the browser part.
